### PR TITLE
✨Add proxy support for node-fetch

### PR DESCRIPTION
**Pull request by thtliife**
## Description

Add proxy agent for node-fetch behind a proxy

Issue: #579 

## Tests

- [x] All tests passed.

### CODE REVIEW

**Comment by carloscuesta** [comment]
Hey @thtliife thank you for opening a PR!

File: src/utils/buildFetchOptions.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+export const buildAgent = (proxy: string = defaultProxy) =>
+  proxy ? new ProxyAgent(proxy) : ''
+
+export const buildFetchOptions = (options: { proxy?: string } = {}) => {
+  const agent = buildAgent(options.proxy)
+  return { agent }
+}
+
+export default buildFetchOptions
```
Comment (carloscuesta) [inline src/utils/buildFetchOptions.js:14]: Can you add a spec file to test this module?

Thanks 🙏🏼

File: src/utils/buildFetchOptions.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// @flow
+import ProxyAgent from 'proxy-agent'
+
+const defaultProxy = process.env.https_proxy || process.env.http_proxy || ''
+
+export const buildAgent = (proxy: string = defaultProxy) =>
+  proxy ? new ProxyAgent(proxy) : ''
+
+export const buildFetchOptions = (options: { proxy?: string } = {}) => {
```
Comment (carloscuesta) [inline src/utils/buildFetchOptions.js:9]: Add a type for the return 
```suggestion
export const buildFetchOptions = (options: { proxy?: string } = {}): { agent: Object } => {
```

**Review by carloscuesta — COMMENTED** [review]
A few details! 😻

File: src/utils/buildFetchOptions.js (hunk 2 above)
Comment (thtliife) [reply src/utils/buildFetchOptions.js:9]: Hi @carloscuesta,
I'll get on those changes this weekend. 
😀

Comment (carloscuesta) [reply src/utils/buildFetchOptions.js:9]: Awesome thanks ❤️

**Comment by AlphaWong** [comment]
useful one. I just want to enforce the git commit lint in my workplace ( with proxy ).
thx both of u

**Comment by thtliife** [comment]
Sorry for the delay... things came up and i couldn't get the specs done as quickly as id like...

**Comment by thtliife** [comment]
This has gotten too untidy... especially with v4 release after the initial pr

Closing, will re-open new PR from fast-forwarded upstream

**Comment by thtliife** [comment]
This has gotten too untidy... especially with v4 release after the initial pr

Closing, will re-open new PR from fast-forwarded upstream

**Pull request closed without merging**